### PR TITLE
Allow content-single.php to be overridden for post formats

### PR DESCRIPTION
**Pull request by grappler**
https://core.trac.wordpress.org/changeset/21491

Fixes: https://github.com/Automattic/_s/issues/675

### CODE REVIEW

File: single.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 		<?php while ( have_posts() ) : the_post(); ?>
 
-			<?php get_template_part( 'content', 'single' ); ?>
+			<?php get_template_part( 'content-single', get_post_format() ); ?>
```
Comment (philiparthurmoore) [inline single.php]: Twenty Fifteen and Twenty Fourteen do it this way:

```php
get_template_part( 'content', get_post_format() );
```
I'd prefer we use the latest way and not the way it was done in Twenty Eleven.

Comment (grappler) [inline single.php]: If we do it this way then we can delete `content-single.php` but will need to copy the differences to `content.php`.

Do you know why Twenty Fifteen and Twenty Fourteen did not include `content-single.php`?

My reasoning for keeping it is that it give a bit more flexibility.

**Comment by grappler** [comment]
I was looking through the code of Twenty Fifteen and Twenty Fourteen and realised that [`search.php`](https://github.com/Automattic/_s/blob/master/search.php#L28) which has the same "issue" as content-single.php.

**Comment by philiparthurmoore** [comment]
@grappler: Is this still a valid PR?

**Comment by grappler** [comment]
Yes, I have come up with a better solution. 

The problem with the current solution is that if there is a `content-image.php` it will not be used in the search and single posts. 

This is what the current default themes are doing:
Twenty Eleven
`single.php`: `get_template_part( 'content-single', get_post_format() );`
`search.php`: `get_template_part( 'content', get_post_format() );`
Twenty Twelve
`single.php`: `get_template_part( 'content', get_post_format() );`
`search.php`: `get_template_part( 'content', get_post_format() );`
Twenty Thirteen
`search.php`: `get_template_part( 'content', get_post_format() );`
`single.php`: `get_template_part( 'content', get_post_format() );`
Twenty Fourteen
`single.php`: `get_template_part( 'content', get_post_format() );`
`search.php`: `get_template_part( 'content', get_post_format() );`
Twenty Fifteen
`single.php`: `get_template_part( 'content', get_post_format() );`
`search.php`: `get_template_part( 'content', 'search' );`

I have created two commit that I can move to thing PR if OK.

Support post format markup on single posts: https://github.com/grappler/_s/commit/167758c337ded38758da7a0384b3c483ebef11ce
Support post format markup on search page: https://github.com/grappler/_s/commit/b680b326dcc55db9c5a1197bedc03314f24c4396

**Comment by karmatosed** [comment]
Are you still interested in proceeding with this? Just doing some housekeeping and don't see any movement.

**Comment by grappler** [comment]
@karmatosed Did you mean me or @philiparthurmoore?

**Comment by philiparthurmoore** [comment]
I suspect that question was for you but my 2 cents: I think it's a great PR, considering that it was done in the default themes for good reason. :smile:

**Comment by grappler** [comment]
@karmatosed I am still interested in working on this. Would love to get some feedback. The latest proposal would be commit: https://github.com/grappler/_s/commit/167758c337ded38758da7a0384b3c483ebef11ce

If it is OK then I could move it to this PR. When this PR is merged I will create a separate PR for the search page. Commit: https://github.com/grappler/_s/commit/b680b326dcc55db9c5a1197bedc03314f24c4396

**Comment by karmatosed** [comment]
Lets maybe get it on this PR so we can have a look in one place and get opinions @grappler ?

**Comment by sarahmonster** [comment]
@grappler: Are you still interesting in working on this? It would be great to update and merge in. :smile_cat:

**Comment by grappler** [comment]
Sorry, yea I will look at on Friday.

**Comment by sarahmonster** [comment]
Thank you! :+1:

**Comment by grappler** [comment]
The PR has been updated and is up-to-date with the master branch.

**Comment by sarahmonster** [comment]
Looks good to me! One quick thing I noticed while testing:

```
if ( is_single() ) {
  the_title( '<h2 class="entry-title">', '</h2>' );
} else {
  the_title( '<h2 class="entry-title"><a href="' . esc_url( get_permalink() ) . '" rel="bookmark">', '</a></h2>' );
}
```

To maintain headers consistent to our existing structure, let's change it to use an H1 for `is_single()`:

```
if ( is_single() ) {
  the_title( '<h1 class="entry-title">', '</h1>' );
} else {
  the_title( '<h2 class="entry-title"><a href="' . esc_url( get_permalink() ) . '" rel="bookmark">', '</a></h2>' );
}
```

@grappler would you be able to do a quick update for us?

**Comment by grappler** [comment]
Thanks, It has been corrected.

**Pull request merged**